pkg/tools/closectx/context.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package closectx provides helper method to create Close/Unregister context in case of Request/Register failure
+package closectx
+
+import (
+	"context"
+	"time"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/clock"
+	"github.com/networkservicemesh/sdk/pkg/tools/extend"
+)
+
+const closeTimeout = 15 * time.Second
```

Comment:
Please don't put magic timer constants in the system.

Reply:
So here comes a problem, what should we use as a timeout in such case? Or should we just make `Close` on a context without any deadline?

Reply:
See below... derive the context from the previous duration of the Request... not from an arbitrary timer value.

Over reliance on (and tuning of) timer values leads to things like: https://github.com/networkservicemesh/cmd-admission-webhook-k8s/pull/18 where we are now seeking to propagate the ability to over tune those timers further and further.

Reply:
Fixed with using timeout taken before the `Request`.